rapids-cmake/cpm/detail/get_proprietary_binary.cmake

```diff
@@ -0,0 +1,78 @@
+#=============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+get_proprietary_binary
+-------------------
+
+.. versionadded:: v22.06.00
+
+Download the associated proprietary binary for the given project based on
+the current CPU target architecture ( x86_64, aarch64, etc )
+
+ .. note::
+  if override => the proprietary entry only in the override will be evaluated
+  if no override => the proprietary entry only in the default will be evaluated
+
+
+#]=======================================================================]
+function(rapids_cpm_get_proprietary_binary package_name version)
+
+  get_property(override_json_data GLOBAL PROPERTY rapids_cpm_${package_name}_override_json)
```

Comment:
Could we create a function `get_package_override` in `cpm/detail` to encapsulate this functionality? The function would be tiny, just checking if the override json file existed and querying it if so, maybe setting a variable in the parent scope or something to indicate what was found.

That would be cleaner than this function needing to know exactly how `rapids_cpm_package_override` transmits information about overridden packages (by setting a global variable). Even if the global is the only way to communicate that to callers of the function, we could at least hide more of the details.

Reply:
I have updated this PR to move the loading from a global property into a function.

Reply:
Does this look better with the refactoring?

Reply:
Yes I like the `get_default_json` and `get_override_json`, that definitely helps!